manifests/common/plumgrid.pp

```diff
@@ -0,0 +1,69 @@
+# This class installs and configures Plumgrid Neutron Plugin.
+
+class openstack::common::plumgrid {
+  $is_controller = $::openstack::profile::base::is_controller
+  $controller_management_address = $::openstack::config::controller_address_management
+  $user = $::openstack::config::mysql_user_neutron
+  $pass = $::openstack::config::mysql_pass_neutron
+  $db_connection = "mysql://${user}:${pass}@${controller_management_address}/neutron"
+
+  if $is_controller {
+    neutron_config {
+      'DEFAULT/service_plugins': ensure => absent,
+    }
+    ->
+    class { '::neutron::plugins::plumgrid':
+      director_server      => $::openstack::config::plumgrid_director_vip,
+      username             => $::openstack::config::plumgrid_username,
+      password             => $::openstack::config::plumgrid_password,
+      admin_password       => $::openstack::config::keystone_admin_password,
+      controller_priv_host => $controller_management_address,
+      connection           => $db_connection,
+    }
+
+    class { '::neutron::agents::metadata':
+      auth_password => $::openstack::config::neutron_password,
+      shared_secret => $::openstack::config::neutron_shared_secret,
+      auth_url      => "http://${controller_management_address}:35357/v2.0",
+      debug         => $::openstack::config::debug,
+      auth_region   => $::openstack::config::region,
+      metadata_ip   => $controller_management_address,
+      enabled       => true,
+    }
+  } else {
+    # forward all ipv4 traffic
+    # this is required for the vms to pass through the gateways public interface
+    sysctl::value { 'net.ipv4.ip_forward': value => '1' }
+
+    file_line { '/etc/libvirt/qemu.conf cgroup_device_acl':
+      path   => '/etc/libvirt/qemu.conf',
+      line   => 'cgroup_device_acl = ["/dev/null","/dev/full","/dev/zero","/dev/random","/dev/urandom","/dev/ptmx","/dev/kvm","/dev/kqemu","/dev/rtc","/dev/hpet","/dev/vfio/vfio","/dev/net/tun"]',
+      match  => 'cgroup_device_acl =',
+      notify => Service['libvirt'],
+    }
+
+    file_line { '/etc/libvirt/qemu.conf clear_emulator_capabilities':
+      path   => '/etc/libvirt/qemu.conf',
+      line   => 'clear_emulator_capabilities = 0',
+      match  => 'clear_emulator_capabilities =',
+      notify => Service['libvirt'],
+    }
+
+    file_line { '/etc/libvirt/qemu.conf user':
+      path   => '/etc/libvirt/qemu.conf',
+      line   => 'user = "root"',
+      match  => 'user =',
+      notify => Service['libvirt'],
+    }
+
+    file_line { '/etc/libvirt/qemu.conf group':
+      path   => '/etc/libvirt/qemu.conf',
+      line   => 'group = "root"',
+      match  => 'group =',
+      notify => Service['libvirt'],
+    }
+
+    Package['libvirt'] -> File_line<| path == '/etc/libvirt/qemu.conf' |>
+  }
+}
```

Comment:
Looking at the documentation in https://wiki.openstack.org/wiki/PLUMgrid-Neutron I don't see anything that mentions changing qemu.conf. This seems to be out of scope for this change. Can you point me to some documentation that says otherwise?